### ISSUE TITLE
Select: onClear is not clickable when using the checkbox variant

### ISSUE THREAD

**Issue by bastilian** [patternfly/patternfly-react#6961]
**Describe the issue. What is the expected and unexpected behavior?**

When using the Select component with the checkbox variant as well as an `onClear` the button to trigger the clearing becomes un-clickable and appears to be covered by the toggle-trigger.

The behaviour can be seen here with items selected: https://codesandbox.io/s/hungry-elgamal-83hccq

**Comment by tlabaj** [comment]
@bastilian the `onClear` prop is intended for use with the `typeahead` variant and not with the checkbox.  That being said we can make un update so the clear button is not rendered for checkbox variant.

@mcarrano do you have any opinion here?

**Comment by kmcfaul** [comment]
@bastilian It looks like the codesandbox `clearSelection` function isn't passed to `onClear` so the button isn't executing that logic currently, and appears to function properly after passing the function in. 

The menu will still toggle closed on a clear though.

**Comment by bastilian** [comment]
Thank you both for taking a look at this!

@tlabaj I assumed since it was in the list of props for the `Select` component it would work regardless of the variant. Is there somewhere in the docs a way to see which props are for which variant? 

@kmcfaul I have a component that builds upon the `Select` component[0], similar to the example, that does have a proper callback updating the state to empty the selection, which shows the same issue.



[0] https://github.com/bastilian/optigit/commit/54e7a765975e9039340de410eee4b1c90369e813#diff-b6d2a12738211a696f70dd7961e5bc79195f375af79c68d103da55b0df789f28R81

**Comment by mcarrano** [comment]
> @mcarrano do you have any opinion here?  

I don't think it was ever intended that we would support a clear button on the checkbox select. I can see this as potentially useful, but it's not typical. @mmenestr any thoughts?

**Comment by mcarrano** [comment]
Per triage meeting on Feb 22, let's go ahead and make this work. Will add to a milestone.

**Comment by kmcfaul** [comment]
@bastilian I've looked further into this but am having trouble reproducing the error (non-clickable onClear button). On this codesandbox I've converted to a functional component and tried a similar setup to your linked code, and the onClear button is clickable and functional. 

Clicking the onClear button will trigger a toggle as well, which will close the menu; is that closer to the issue you are experiencing? Or is the button not clickable at all?

https://codesandbox.io/s/adoring-pateu-9sqzr9?file=/index.js
